go/libraries/doltcore/sqle/dprocedures/dolt_push.go

```diff
@@ -83,7 +83,7 @@ func doDoltPush(ctx *sql.Context, args []string) (int, error) {
 	if err != nil {
 		switch err {
 		case doltdb.ErrUpToDate:
-			return cmdSuccess, nil
+			return cmdSuccess, err
```

Comment:
why was this nil before? Seems like this could change some tests, but I suspect you are working through that.

Reply:
it makes some bats tests a little more wordy, needing to catch the up to date error in places where we didn't have to catch anything before. will probably also need a minor version bump for this change since we're returning an error where we didn't before.

Reply:
But being up-to-date means that it's a no-op, right?